nvflare/private/fed/client/client_executor.py

```diff
@@ -214,6 +185,18 @@ def start_app(
         )
         thread.start()
 
+    def _get_job_launcher(self, job_id, client, job_meta: dict) -> dict:
+        launch_image = extract_job_image(job_meta, client.client_name)
```

Comment:
This core code shouldn't even know anything about "image". If you fire an event here, each launcher will simply make their decision based on the job meta data.

Reply:
changed to event firing.